tobler/util/raster_tools.py

```diff
@@ -0,0 +1,63 @@
+"""tools for dealing with rasters."""
+
+import ast
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio as rio
+from rasterio import features
+from shapely.geometry import shape
+
+
+def _parse_geom(geom_str):
+    return shape(ast.literal_eval(geom_str))
+
+
+def extract_raster_features(gdf, raster_path, pixel_types=None, nodata=255):
+    """Generate a geodataframe from raster pixels.
```

Comment:
This is very vague, what is it doing? Is it rasterising like [here](https://rasterio.readthedocs.io/en/latest/topics/features.html?highlight=rasterize#vector-features)? Converting pixel squares into a vector geo-table? I think the docstring needs a bit more verbose to explain what it does.

Reply:
no, its doing the opposite--its featurizing (the first example on that page, not the second)

Reply:
the description in the `Returns` section has a bit more info, but i can still clarify the docstring further